alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php

```diff
@@ -156,30 +158,64 @@ private function getOutputHash($o)
 		return $this->eTagHash;
 	}
 	
-	private function getAutoEmbedCode()
+	private function getAutoEmbedCode($targetId = null)
 	{
-		$config = json_encode($this->getRequestParameter("config"));
-		$entry_id = $this->getRequestParameter('entry_id');
-		$autoEmbedCode = "\n var player; var ovpProvider = new Providers.OvpProvider($this->partnerId,\"\",$config);\n" .
-			"\t    ovpProvider.getConfig(\"" . $entry_id . "\",$this->uiconfId).then(config => {\n" .
-			"\t    player = Playkit.playkit(config);\n" .
-			"\t }, \n" .
-			"\t err => {\n" .
-			"\t    console.log(err)\n" .
-			"\t})\n";
+		if (!$targetId){
+			$targetId = $this->getRequestParameter('targetId');
+		}
+		if (!$targetId){
+			KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, "Player target ID not defined");
+		}
+		$entry_id = $this->getRequestParameter(self::ENTRY_ID_PARAM_NAME);		
+		if (!$entry_id){
+			KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, "Entry ID not defined");
+		}
+		$config = $this->getRequestParameter(self::CONFIG_PARAM_NAME);		
+		//enable passing nested config options
+		foreach ($config as $key=>$val){
+			$config[$key] = json_decode($val);
+		}
+
+		$config["partnerId"] = $this->partnerId;		
+		$config["uiConfId"] = $this->uiconfId;
+		
+		try{
+			$config = json_encode($config);		
+		} catch (Exception $ex) {
```

Comment:
json_encode does not throw , on failure it will return false.

Reply:
fixed